Marlin/src/module/configuration_store.cpp

```diff
@@ -1180,13 +1187,22 @@ void MarlinSettings::postprocess() {
         for (uint16_t q = mesh_num_x * mesh_num_y; q--;) EEPROM_READ(dummy);
       #endif // MESH_BED_LEVELING
 
-      _FIELD_TEST(zprobe_zoffset);
-
       #if !HAS_BED_PROBE
         float zprobe_zoffset;
       #endif
-      EEPROM_READ(zprobe_zoffset);
-
+      #if ENABLED(INDEPENDENT_Z_OFFSETS) && ENABLED(DUAL_X_CARRIAGE)
+        _FIELD_TEST(x1_zprobe_zoffset);
+        EEPROM_READ(x1_zprobe_zoffset);
+        _FIELD_TEST(x2_zprobe_zoffset);
+        EEPROM_READ(x2_zprobe_zoffset);
+        if(active_extruder==1)
+          zprobe_zoffset = x2_zprobe_zoffset;
+        else
+          zprobe_zoffset = x1_zprobe_zoffset;
```

Comment:
For consistency, if you're using `active_extruder==1` in the other cases, it should also be used here. Comparison to zero also produces smaller code.

Keywords like `if` must be followed by a single space.